src/nix/all_deps.rs

```diff
@@ -0,0 +1,75 @@
+use std::fmt;
+
+use anyhow::{bail, Result};
+
+use super::devour_flake::DrvOut;
+
+/// Nix derivation path
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone, Hash)]
+pub struct Drv(pub String);
+
+/// Encompasses both derivation and derivation output paths
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone, Hash)]
+pub enum DrvPaths {
+    DrvOut(DrvOut),
+    Drv(Drv),
+}
+
+impl fmt::Display for DrvPaths {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            DrvPaths::DrvOut(drv_out) => write!(f, "{}", drv_out.0),
+            DrvPaths::Drv(drv) => write!(f, "{}", drv.0),
+        }
+    }
+}
+
+/// Query the deriver of an output path
```

Comment:
This comment doesn't say anything new in addition to what the function name already implies. Can it be rewritten so that a Rust developer familiar with Nix but not yet familiar with `nix-store ...` CLI can understand what it does?